corehq/apps/users/audit/change_messages.py

```diff
@@ -1,91 +1,329 @@
-class UserChangeMessage(object):
+from enum import Enum
+from django.utils.translation import ugettext as _
+
+
+class Change(Enum):
+    SET = 'set'
+    RESET = 'reset'
+    REMOVE = 'remove'
+    ADD = 'add'
+
+
+class UserChangeMessageV1(object):
+    """
+    Each change message to follow the structure
+    {
+        "field": {
+            "change": change_details
+        }
+    }
+    field: could be domain, phone_numbers etc
+    change: either 'set' / 'reset' / 'remove' / 'add'
+    change_details: a Boolean or
+                    a dict with details as needed for the change
+                    a list of a dicts as needed for the change
+    """
     @staticmethod
     def program_change(program):
```

Comment:
I like how this class can enforce that callers include the necessary information and format